### PR TITLE
Update rubygems and add a devcontainer

### PR DESCRIPTION
**Pull request by razzeee**
This updates all gems, so that it can build again with the current versions of ruby and rubygem.

It also adds a devcontainer, so that people can run it without installing anything ruby on their system.

### CODE REVIEW

**Comment by razzeee** [comment]
Closes https://github.com/flatpak/flatpak.github.io/issues/506

File: Gemfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 source 'https://rubygems.org'
 
-gem "middleman", "~> 3"#, github: "middleman/middleman", branch: "v3-stable"
-gem "middleman-matomo", github: "barthalion/middleman-matomo"
+gem "middleman", "~> 4"#, github: "middleman/middleman", branch: "v3-stable"
+gem 'matomo-middleman'
```
Comment (razzeee) [inline Gemfile:4]: This seems to work fine, as far as I can tell without deploying it. Locally everything seems to be added correctly.

But I'm not sure what the changes in @barthalion fork were

File: Gemfile [hunk 2]
```diff
@@ -16,14 +16,14 @@ gem "sass" #3.2.12
 # gem "middleman-syntax"
 
 #deployment
-gem "middleman-deploy" #, "~> 2.0.0.pre.alpha"
+gem "middleman-deploy", "~> 2.0.0.pre.alpha"
```
Comment (razzeee) [inline Gemfile:19]: Based on https://github.com/karlfreeman/middleman-deploy/issues/132#issuecomment-337524002 doesn't work without for me

File: Gemfile (hunk 1 above)
Comment (barthalion) [reply Gemfile:4]: I don't actually remember at this point.

Comment (razzeee) [reply Gemfile:4]: Guess we figured out why :D

**Pull request merged**